pytests/test_container_to_args.py

```diff
@@ -325,3 +325,106 @@ async def test_env_file_obj_optional(self):
                 "busybox",
             ],
         )
+
+    async def test_gpu(self):
```

Comment:
This test is actually 4 tests. Could you split it. It's ok if the preamble

```
        c = create_compose_mock()

        cnt = get_minimal_container()
        cnt["command"] = ["nvidia-smi"]
        cnt["deploy"] = {"resources": {"reservations": {"devices": [{}]}}}
```

repeats in each test.